pkg/apiserver/handlers_test.go

```diff
@@ -151,6 +151,15 @@ func TestGetAPIRequestInfo(t *testing.T) {
 		{"GET", "/namespaces", "list", "", "", "namespaces", "", "Namespace", "", []string{"namespaces"}},
 		{"GET", "/namespaces/other", "get", "", "other", "namespaces", "", "Namespace", "other", []string{"namespaces", "other"}},
```

Comment:
So, the namespace etcd tests kept passing because I had (incorrectly) updated them to have an empty context. I updated the namespace/etcd_test contexts to match reality:

* create, list, watch calls have an empty namespace in the context
* get, update, delete calls have the current namespace in the context

Take a look at the namespace/etcd changes... I'm not thrilled with them, but I think the registry should be explicit about discarding the namespace from the context by actually doing it, rather than just ignoring it in it's key-building function.